src/global/variables/RandomDouble.java

```diff
@@ -5,27 +5,57 @@
 import java.util.ArrayList;
 import java.util.HashMap;
 
-public class RandomDouble implements Clause, RandomVariable<Double> {
+public class RandomDouble extends Clause implements RandomVariable<Double> {
     private double lower, upper;
     private int precision = 16;  // max double precision
 
     public RandomDouble(double lower, double upper) {
+        super();
+        this.lower = lower;
+        this.upper = upper;
+    }
+
+    public RandomDouble(double lower, double upper, String name) {
+        super(name);
         this.lower = lower;
         this.upper = upper;
     }
 
     public RandomDouble(int lower, int upper) {
+        super();
+        this.lower = lower;
+        this.upper = upper;
+    }
+
+    public RandomDouble(int lower, int upper, String name) {
+        super(name);
         this.lower = lower;
         this.upper = upper;
     }
 
     public RandomDouble(double lower, double upper, int precision) {
+        super();
+        this.lower = lower;
+        this.upper = upper;
+        this.precision = precision;
+    }
+
+    public RandomDouble(double lower, double upper, int precision, String name) {
+        super(name);
         this.lower = lower;
         this.upper = upper;
         this.precision = precision;
     }
 
     public RandomDouble(int lower, int upper, int precision) {
```

Comment:
The `int` ones could just call `RandomDouble(all of their args)` and then it'll overload and call the double constructors by default, right?

Reply:
It won't call the double unless atleast one of the values in casted explicitly

Reply:
So like
```java
public RandomDouble(int lower, int upper, int precision) {
     RandomDouble(lower, upper, precision, name);
}```
doesn't work here?